### PR TITLE
[@mantine/core] SegmentedControl - migrate to useResizeObserver hook

### LINKED ISSUE #585: SegmentedControl's value is shifted to the left after opening it in a Modal
### What package has an issue

@mantine/core

### Describe the bug

Create a Modal with a SegmentedControl inside of it, and store the SegmentedControl's value with `useState`. Click the SegmentedControl to change it's value. Close the modal, and open it again. Now the displayed handle is shifted to the left by a bit (I'm not sure if it's called a "handle", but [this](https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/components/SegmentedControl/SegmentedControl.tsx#L157-L163) is what I mean).

In this picture I have "Value 4" selected: 

Here's a video where I show the steps to reproduce the bug:

https://user-images.githubusercontent.com/38920928/147412465-305fca23-9b90-4e0a-9a73-98dfb1e7df1c.mp4

### In which browser did the problem occur

Firefox, Chrome, Edge

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/priceless-bassi-yozl1

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

_No response_

### Possible fix

_No response_

### PR DESCRIPTION
**Pull request by achmurali**
Migrating SegmentedControl to useResizeObserver hook..

Update - 
This PR also fixes #585.

### CODE REVIEW

File: src/mantine-core/src/components/SegmentedControl/SegmentedControl.tsx [hunk 1]
```diff
@@ -101,27 +101,26 @@ export function SegmentedControl<T extends string = string>({
...
-          setTimeout(() => {
-            setShouldAnimate(true);
-          }, 4);
-        }
+    if (_value in refs.current && ref.current) {
+      const element = refs.current[_value];
+      const elementRect = element.getBoundingClientRect();
+
+      // Scaled value to undo transforms applied
+      const scaledValue = element.offsetWidth / elementRect.width;
```
Comment (rtivital) [inline src/mantine-core/src/components/SegmentedControl/SegmentedControl.tsx:112]: Please also add a story that will have scale toggle (like in sandbox shared in issue – https://codesandbox.io/s/romantic-breeze-q3e99?file=/src/App.js)

**Comment by rtivital** [comment]
Were you able to fix the issue?

**Comment by achmurali** [comment]
> Were you able to fix the issue?

Yeah @rtivital , after a lot of toil😅

**Comment by rtivital** [comment]
Resize observer does not work

https://user-images.githubusercontent.com/10353856/148515428-bd49af5d-1551-46be-b812-a4604a493b53.mov

**Comment by rtivital** [comment]
Thanks, it works great!

**Pull request merged**